extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -3,51 +3,76 @@
  */
 package io.deephaven.parquet.base;
 
+import io.deephaven.UncheckedDeephavenException;
 import io.deephaven.parquet.base.util.Helpers;
+import io.deephaven.parquet.base.util.SeekableChannelContext;
 import io.deephaven.parquet.base.util.SeekableChannelsProvider;
 import org.apache.parquet.format.*;
 import org.apache.parquet.format.ColumnOrder;
 import org.apache.parquet.format.Type;
 import org.apache.parquet.schema.*;
 
 import java.io.ByteArrayInputStream;
+import java.io.File;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.*;
 
+import static io.deephaven.parquet.base.util.SeekableChannelsProvider.convertToURI;
+
 /**
- * Top level accessor for a parquet file
+ * Top level accessor for a parquet file which can read both from a file path string or a CLI style file URI,
+ * ex."s3://bucket/key".
  */
 public class ParquetFileReader {
     private static final int FOOTER_LENGTH_SIZE = 4;
     private static final String MAGIC_STR = "PAR1";
     static final byte[] MAGIC = MAGIC_STR.getBytes(StandardCharsets.US_ASCII);
+    public static final String FILE_URI_SCHEME = "file";
 
     public final FileMetaData fileMetaData;
     private final SeekableChannelsProvider channelsProvider;
-    private final Path rootPath;
+
+    /**
+     * If reading a single parquet file, root URI is the URI of the file, else the parent directory for a metadata file
+     */
+    private final URI rootURI;
     private final MessageType type;
 
-    public ParquetFileReader(final String filePath, final SeekableChannelsProvider channelsProvider)
+    public ParquetFileReader(final String source, final SeekableChannelsProvider channelsProvider)
             throws IOException {
-        this.channelsProvider = channelsProvider;
-        // Root path should be this file if a single file, else the parent directory for a metadata
-        // file
-        rootPath =
-                filePath.endsWith(".parquet") ? Paths.get(filePath) : Paths.get(filePath).getParent();
+        this(convertToURI(source), channelsProvider);
+    }
 
+    public ParquetFileReader(final URI parquetFileURI, final SeekableChannelsProvider channelsProvider)
+            throws IOException {
+        this.channelsProvider = channelsProvider;
+        if (!parquetFileURI.getRawPath().endsWith(".parquet")) {
+            // Construct a new URI for the parent directory
+            try {
+                rootURI = new URI(parquetFileURI.getScheme(), parquetFileURI.getHost(),
+                        new File(parquetFileURI.getPath()).getParent(), parquetFileURI.getRawFragment());
```

Comment:
This seems like it might be a mis-use of the `File` API. I wonder if, for example, it might not work on OSes with a different path separator?

Reply:
I have simplified the logic by adding an additional check for it to be a file URI.